engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -894,6 +894,175 @@ SELECT ?obsId {
         expect(bindings2).toMatchObject(expectedResult);
       });
     });
+
+    describe('initialbindings', () => {
+      let initialBindings: Bindings;
+      let sourcesValue1: string;
+
+      beforeEach(() => {
+        initialBindings = BF.bindings([
+          [ DF.variable('a'), DF.namedNode('http://example.org/test#testBinding') ],
+        ]);
+        sourcesValue1 = `
+          @prefix ex: <http://example.org/test#> .
+          
+          ex:testBinding
+            ex:property "testProperty" .
+          `;
+      });
+
+      it('should consider the initialbindings in the bound function', async() => {
+        const context: QueryStringContext = {
+          sources: [
+            {
+              type: 'serialized',
+              value: sourcesValue1,
+              mediaType: 'text/turtle',
+            },
+          ],
+          initialBindings,
+        };
+
+        const expectedResult = [
+          [
+            [ DF.variable('a'), DF.namedNode('http://example.org/test#testBinding') ],
+          ],
+        ];
+
+        const bindings = (await arrayifyStream(await engine.queryBindings(`
+        PREFIX ex: <http://example.org/test#>
+        
+        SELECT $a WHERE {
+          {
+            FILTER (bound($a))
+          }
+          $a ex:property "testProperty" .
+          FILTER (bound($a)) .
+        }
+          `, context))).map(binding => [ ...binding ].sort(([ var1, _c1 ], [ var2, _c2 ]) => var1.value.localeCompare(var2.value)));
+
+        expect(bindings).toMatchObject(expectedResult);
```

Comment:
Instead of sorting, you can use `toEqualBindingsStream` like in the other tests.